app/models/training_data/code_tags_sample.rb

```diff
@@ -0,0 +1,52 @@
+class TrainingData::CodeTagsSample < ApplicationRecord
+  serialize :tags, JSON
+  serialize :files, JSON
+
+  belongs_to :track
+  belongs_to :exercise, optional: true
+  belongs_to :solution, optional: true
+  belongs_to :locked_by, class_name: "User", optional: true
+
+  enum dataset: { training: 0, validation: 1 }
+  enum status: { untagged: 0, machine_tagged: 1, human_tagged: 2, community_checked: 3, admin_checked: 4 }
+
+  scope :unlocked, -> { where(locked_until: nil).or(self.where('locked_until < NOW()')) }
+
+  before_validation on: :create do
+    self.uuid = SecureRandom.compact_uuid unless self.uuid
+    self.dataset = (rand < 0.9 ? :training : :validation) unless dataset
+
+    self.exercise_id = solution.exercise_id unless exercise_id
+    self.track_id = exercise.track_id unless track_id
+  end
+
+  def to_param = uuid
+
+  def dataset = super.to_sym
+  def status = super.to_sym
+
+  def checked?
+    status == :human_tagged || status == :admin_tagged
+  end
```

Comment:
This looks wrong to me. What is `checked?` - I'd presume it should be at least one of the `_checked` statuses.

Reply:
Well, you tell me :) https://github.com/exercism/website/blame/5404db7a3d591b676b5a3b7ce91cee880aa6fd70/app/models/training_data/code_tags_sample.rb

Reply:
Let's rename this to `safe_to_override?` and also add `community_checked` as an `||`